src/internal/connector/graph/service.go

```diff
@@ -25,6 +25,28 @@ type QueryParams struct {
 	Credentials   account.M365Config
 }
 
+var _ Servicer = &Service{}
+
+type Service struct {
+	adapter msgraphsdk.GraphRequestAdapter
+	client  msgraphsdk.GraphServiceClient
+}
+
+func NewService(adapter *msgraphsdk.GraphRequestAdapter) Service {
+	return Service{
+		adapter: *adapter,
+		client:  *msgraphsdk.NewGraphServiceClient(adapter),
```

Comment:
if we return pointers to these anyway in `Adapther()` and `Client()` is there a reason to store a copy of them instead of a pointer to them?

Reply:
so, here's the question: my inclination is to make all of this a value copy.  Are there strong preferences towards passing Service as a pointer instead?

Reply:
mmm I think it would be good to see what the functions that consume service/adapter need, especially in the Graph SDK. If the SDK mostly uses pointers then we should return pointers so we don't have to take the address of stuff all over the place. Beyond that I don't really have a preference